src/layout/IHyprLayout.cpp

```diff
@@ -548,14 +555,43 @@ void IHyprLayout::onMouseMove(const Vector2D& mousePos) {
     if (g_pInputManager->currentlyDraggedWindow.expired())
         return;
 
-    const auto DRAGGINGWINDOW = g_pInputManager->currentlyDraggedWindow.lock();
+    const auto  DRAGGINGWINDOW = g_pInputManager->currentlyDraggedWindow.lock();
+    static auto PDRAGTHRESHOLD = CConfigValue<Hyprlang::INT>("input:drag_threshold");
 
     // Window invalid or drag begin size 0,0 meaning we rejected it.
     if ((!validMapped(DRAGGINGWINDOW) || m_vBeginDragSizeXY == Vector2D())) {
         g_pKeybindManager->changeMouseBindMode(MBIND_INVALID);
         return;
     }
 
+    // Yoink window here instead if using drag_threshold
+    if (*PDRAGTHRESHOLD > 0 && !g_pInputManager->m_bDragThresholdReached) {
+        if (m_vBeginDragXY.distance(mousePos) <= *PDRAGTHRESHOLD && m_vBeginDragXY == m_vLastDragXY)
+            return;
+
+        if (DRAGGINGWINDOW->isFullscreen()) {
+            Debug::log(LOG, "Dragging a fullscreen window");
+            g_pCompositor->setWindowFullscreenInternal(DRAGGINGWINDOW, FSMODE_NONE);
+        }
+
+        const auto PWORKSPACE = DRAGGINGWINDOW->m_pWorkspace;
```

Comment:
looks convoluted and duped. Why not extract the logic for "picking up" the window to a method and call that?

Reply:
I added a new method that both updates the drag origin related values and tries to yoink the window if the threshold is reached/disabled. Pretty tilted right now, don't know if anything makes sense or if it's in the right place.

At least using move and resize with a big threshold feels very smooth now that the DragXY values are updated again when the threshold is reached...